src/main/xerus/monstercat/api/Playlist.kt

```diff
@@ -86,4 +101,181 @@ object Playlist {
 			else add(track)
 		}
 	}
+}
+
+object PlaylistManager {
+	suspend fun loadPlaylist(apiConnection: APIConnection){
+		val tracks = apiConnection.getTracks()
+		tracks?.forEachIndexed { index, track ->
+			val found = Cache.getTracks().find {
+				it.id == track.id
+			}
+			if (found != null)
+				tracks[index] = found
+			else
+				tracks.removeAt(index)
+		}
+		if (tracks != null && tracks.isNotEmpty()) {
+			if (Player.player?.status != MediaPlayer.Status.DISPOSED)
+				Player.reset()
+			Playlist.setTracks(tracks)
+		}
+	}
+
+	/** Opens the playlist manager dialog
+	 * Allows to load, save, and manage playlists stored on Monstercat.com
+	 */
+	fun playlistDialog(){
```

Comment:
Why a dialog and not a tab? Dialogs should be used sparingly.

Reply:
A full tab for this ? Are you serious ? This is just a small feature based around Playlist tab. It won't work without it.

Reply:
No ^^ was just a first thought, but doesn't make sense